lib/templates/src/components/InteractiveWelcome.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+  <div>
+    This screen sets up the socket connection for the interactive experiment.
+    Previously, the participant would need to agree to the terms of the
+    interactive experiment before continuing to the lobby. However, if the terms
+    are already agreed to in the previous welcome screen, you may want to
+    advance this screen automatically.
+  </div>
+</template>
+
+<script>
+import { initializeExperiment } from '@/socket.js';
+
+export default {
+  name: 'InteractiveWelcome',
+  inject: {
+    $exp: {
+      from: 'experiment',
+      default: () => ({})
+    }
+  },
+  watch: {
+    '$store.state.interactiveExperiment.experimentAvailable': function (value) {
```

Comment:
I tried to find ways to simplify it as you mentioned, though it would actually involve importing the store and writing out a separate `computed` part in the component. Not sure if that would actually add overhead instead. You may compare the approaches at https://stackoverflow.com/questions/43270159/vue-js-2-how-to-watch-store-values-from-vuex

Reply:
Yeaah, I see what you mean :/